float32 dist_x
float32 dist_y
float32 dist_abs